custom/icds_reports/static/js/directives/awc-reports/awc-reports.directive.js

```diff
@@ -2084,6 +2084,102 @@ function AwcReportsController($scope, $http, $location, $routeParams, $log, DTOp
+
+    vm.getMobileRequestParams = function (step) {
+        var requestParams = JSON.parse(JSON.stringify($location.search()));
+        for (var i = 0; i < vm.awcReportTableData[step].length; i++) {
+            var entry = vm.awcReportTableData[step][i];
+            requestParams['columns[' + i + '][data]'] = entry['mData'];
+        }
+        requestParams['order[0][column]'] = vm.sortingColumn;
+        requestParams['order[0][dir]'] = vm.dataSortingDirection;
+        requestParams['start'] = vm.requestDataStartingFrom;
+        requestParams['length'] = 10;
+
+        return requestParams;
+    };
+
+    vm.generateSortableKpiData = function (data) {
+        var tableData = vm.awcReportTableData[vm.step];
+        var existingDataLength = vm.sortableInputKpiData.length;
+        for (var i = existingDataLength; i < (data.length + existingDataLength); i++) {
+            vm.sortableInputKpiData[i] = {};
+            vm.sortableInputKpiData[i]['cardHeading'] = tableData[0]['value']('', '', data[i - existingDataLength]);
+            vm.sortableInputKpiData[i]['attributes'] = [];
+            var cardData = data[i - existingDataLength];
+            for (var j = 1; j < tableData.length; j++) {
+                var kpiObject = {
+                    'heading' : tableData[j]['title'],
+                    'value' : tableData[j]['value']('', '', cardData),
+                    'isTheSortedColumn' : j === vm.sortingColumn,
+                    'sortingDirection' : vm.dataSortingDirection,
+                    'tooltip' : tableData[j]['tooltipType'] ?
+                        vm.getPopoverContent(cardData['recorded_weight'], cardData['recorded_height'],
+                            cardData['age_in_months'], tableData[j]['tooltipType']) : null,
+                };
+                // adding the sorted column as first element (unshift adds elements to array at beginning, push at end)
+                if (kpiObject['isTheSortedColumn']) {
+                    vm.sortableInputKpiData[i]['attributes'].unshift(kpiObject);
+                } else {
+                    vm.sortableInputKpiData[i]['attributes'].push(kpiObject);
+                }
+            }
+        }
+    };
+
+    jQuery(function($) {
+        $('#summaryList').bind('scroll', function () {
+            // this function runs when end of summary list is reached and if totalRecords are not yet requested
+            // Also requests are prevented for initial requests
+            // Reference: http://jsfiddle.net/doktormolle/w7X9N/
+            if (isMobile && ($(this).scrollTop() + $(this).innerHeight() >= $(this)[0].scrollHeight) &&
+                ((vm.requestDataStartingFrom + 10) < vm.totalNumberOfEntries) && vm.sortableInputKpiData.length) {
+                vm.requestDataStartingFrom += 10;
+                vm.getDataForStep(vm.step);
+            }
+        })
+    });
```

Comment:
Not thrilled with the amount of similarity with service delivery dashboard. Did you think about / attempt to deduplicate at all? I haven't thought too hard about how that would be possible or best done - maybe some kind of datatables base class?

Reply:
I will try extracting common code.